src/types/package.go

```diff
@@ -28,11 +28,12 @@ type ZarfMetadata struct {
 
 // ZarfBuildData is written during the packager.Create() operation to track details of the created package.
 type ZarfBuildData struct {
-	Terminal     string `json:"terminal"`
-	User         string `json:"user"`
-	Architecture string `json:"architecture"`
-	Timestamp    string `json:"timestamp"`
-	Version      string `json:"version"`
+	Terminal     string   `json:"terminal"`
+	User         string   `json:"user"`
+	Architecture string   `json:"architecture"`
+	Timestamp    string   `json:"timestamp"`
+	Version      string   `json:"version"`
+	Migrations   []string `json:"migrations"`
```

Comment:
When printing the zarf.yaml when deploying packages to get validation from the user that the deployment is correct the depreciated contents will be printed as well as the migrated contents. 

For this specific depreciation that might be ok (although still a little confusing), but I could guess that being less clear for some random potential future depreciation.

Thoughts on adding some utility function that simply removes the old deprecated content (if the package has already been migrated) to make the validation output make more sense?

I think this could be in a separate PR but this felt like a good place to start thinking about it.

Reply:
Adding a migration tool in a future pr makes sense to me. We already warn users about the deprecation when we show the yaml though, is that not enough?

Reply:
We don't show that warning when deploying a package that has had deprecated features already migrated.

In the following screenshot shows me deploying a package, in t he `Deploy this Zarf package?` conformation I am being shown both the scripts and the actions but I'm not seeing a warning that there has been a migration b/c we skipped that process for deploy due to the migration tag existing in the build data.
<img width="1077" alt="Screen Shot 2023-01-24 at 11 10 10 AM" src="https://user-images.githubusercontent.com/18587823/214346528-78791fae-63f6-4439-9ccb-6addf84acc40.png">

Reply:
Someone could see this prompt and easily think the `touch test-deprecated-deploy-before-hook.txt` command is going to be run twice.

Reply:
We could have the migration delete the scripts from the component if it sees that scripts to actions has taken place right?

https://github.com/defenseunicorns/zarf/blob/jeff-doing-stupid-jeff-things-smarter-migrations/src/pkg/packager/deprecated/scripts-to-actions.go